### PR TITLE
Use unpadded base64 strings for IDs and secrets

### LINKED ISSUE #18: nicer RoomID
Since users are in contact with URLs which contain the roomID, it would be nice to only use alphanumeric chars, and maybe have a shorter RoomID.

### PR DESCRIPTION
**Pull request by abustany**
This fixes #18 (at least partly :) )

### CODE REVIEW

File: e2e/everything_test.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ Scenario('Test everything', async (I) => {
   identifyAs(I, user1name)
 
   const url = await I.grabCurrentUrl();
-  const roomId = new URL(url).searchParams.get('id');
+  const roomId = new URL(url).searchParams.get('roomId');
```
Comment (wowawiwa) [inline e2e/everything_test.ts:35]: I'd keep `id` because it's shorter

**Comment by abustany** [comment]
The revert is maybe a bit "brutal" yes, you're right :) I'll amend it so that it removes only the incorrect stripping of the last two characters in generated IDs. This only works because the base64 parser on the server side is not too strict, and potentially discards some data.

**Comment by wowawiwa** [comment]
What stripping is incorrect I don't get it? If it's the `==`, they are added again in relevant places.

**Comment by abustany** [comment]
Well stripping the last two characters unconditionally only works as long as the client ID length is 16 bytes, I would not encode this assumption in a hidden way in other parts of the code. With this patch (assuming I fix it to keep `id` instead of `roomId` etc) I'm "officially" switching both the backend and the frontend to unpadded strings, so we don't need to unpad and repad on the frontend side in many places.

**Comment by wowawiwa** [comment]
Yes that's all clear, but it is not "incorrect" nor is it related to a parser not being strict or discarding data as far as understand until now, nor is it "in other parts of the code". On the other hand, it's nicely isolated, waiting for this backend change to happen.

Instead of reverting, you can just change the read and write functions that are following each other there:
https://github.com/abustany/goretro/pull/55/files#diff-1142a7bb977146abdfcc76f264af3125L138
https://github.com/abustany/goretro/pull/55/files#diff-1142a7bb977146abdfcc76f264af3125L148

**Comment by abustany** [comment]
OK I fixed the patch to only change the `==` stripping/adding :ok_hand:

**Comment by wowawiwa** [comment]
👌 ✨ 

Just in case because I saw `TrimRight`, can it possibly cause a bug if the string would end with `===` rather than `==`?

**Comment by abustany** [comment]
= in base64 is a padding character so that the length of the decoded data is a multiple of 8 bits. Now that the parser is configured in unpadded mode, we can safely remove all the padding. Also, in a valid padded base64 string there should never be more than two = signs at the end.

**Pull request merged**